examples/csharp/src/client.cs

```diff
@@ -0,0 +1,136 @@
+using RestSharp;
+using System.Collections.Generic;
+
+public class Client
+{
+    protected string _endpoint;
+    protected bool _selfSigned;
+    protected Dictionary<string, string> _globalHeaders;
+
+    public Client()
+    {
+        _endpoint = "https://appwrite.io/v1";
+        _globalHeaders = new Dictionary<string, string> { "content-type"="", "x-sdk-version"= "appwrite:csharp:" };
+        _selfSigned = false;
+    }
+
+    public SetSelfSigned(bool status = true)
+    {
+        _selfSigned = status;
+    }
+
+    public SetEndpoint(string endpoint)
+    {
+        _endpoint = endpoint;
+    }
+
+    public AddHeader(string key, string value)
+    {
+        if (_globalHeaders.ContainsKey(key.ToLower()))
+        {
+            _globalHeaders[key.ToLower()] = value.ToLower();
+        }
+        else
+        {
+            _globalHeaders.Add(key, value.ToLower());
+        }
+    }
+
+    public SetProject(string value)
+    {
+        if (_globalHeaders.ContainsKey(key.ToLower()))
+        {
+            _globalHeaders["x-appwrite-project"] = value.ToLower();
+        }
+        else
+        {
+            _globalHeaders.Add("x-appwrite-project", value.ToLower());
+        }
+    }
+
+    public SetKey(string value)
+    {
+        if (_globalHeaders.ContainsKey(key.ToLower()))
+        {
+            _globalHeaders["x-appwrite-key"] = value.ToLower();
+        }
+        else
+        {
+            _globalHeaders.Add("x-appwrite-key", value.ToLower());
+        }
+    }
+
+    public SetLocale(string value)
+    {
+        if (_globalHeaders.ContainsKey(key.ToLower()))
+        {
+            _globalHeaders["x-appwrite-locale"] = value.ToLower();
+        }
+        else
+        {
+            _globalHeaders.Add("x-appwrite-locale", value.ToLower());
+        }
+    }
+
+    public SetMode(string value)
+    {
+        if (_globalHeaders.ContainsKey(key.ToLower()))
+        {
+            _globalHeaders["x-appwrite-mode"] = value.ToLower();
+        }
+        else
+        {
+            _globalHeaders.Add("x-appwrite-mode", value.ToLower());
+        }
+    }
+
+    public Call(string method, string path="", Dictionary<string, string> headers = null, Dictionary<string, string> paramsIn = null)
+    {
+        var restClient = new RestClient(_endpoint);
+        var request = new RestRequest(path)
+        {
+            Method = method
+        };
+
+        if (headers != null && headers.Count > 0)
+        {
+            foreach (KeyValuePair<string, string> kvp in headers)
```

Comment:
So if this Client is reused many times it will keep adding headers to the globalHeaders and send these in subsequent requests? Don't you need to reset the headers to a known state when a REST request is finished?